anonymise/src/lib.rs

```diff
@@ -0,0 +1,328 @@
+//! ## Court document package anonymiser library
+//!
+//! This library contains common code shared between the anonymiser script and the lambda.
+use clap::Parser;
+use docx_rs::*;
+use flate2::{read::GzDecoder, write::GzEncoder, Compression};
+use serde_json::{json, Value};
+use sha256::try_digest;
+
+use std::fs::{remove_file, DirEntry};
+use std::io::ErrorKind;
+use std::{fs, fs::File, io, io::Error, io::Read, path::Path, path::PathBuf};
+use tar::{Archive, Builder};
+
+/// # A struct representing the input arguments
+#[derive(Parser)]
+#[clap(name = "anonymiser")]
+pub struct Opt {
+    /// Input folder
+    #[clap(long, short, value_parser)]
+    pub input: String,
+
+    /// Output folder
+    #[clap(long, short, value_parser)]
+    pub output: String,
+}
+
+/// # Package processor
+/// This takes an output directory path and a path to a tar.gz file as input and anonymises them with the following steps:
+///
+/// * It replaces Contact-Email and Contact-Name with XXXXXXX
+/// * It generates a new docx file which only contains the name of the judgment.
+/// * It updates the checksum field with the calculated checksum of the new docx file.
+/// * It renames the folder and metadata file from TDR-xxx to TST-xxx.
+/// * It creates a new tar.gz folder in the output directory.
+/// * It deletes the uncompressed folder in the output directory.
+pub fn process_package(dir_output: &PathBuf, file: &PathBuf) -> Result<PathBuf, Error> {
```

Comment:
1. process_package is ok. We've already got anonymise everywhere 
2. You asked this last time :smile: It's tricky because lots of variables are used throughout the function so you end up with the smaller functions having to return multiple variables which looks messy.

Reply:
1. process_package is ok. We've already got anonymise everywhere 
2. You asked this last time :smile: It's tricky because lots of variables are used throughout the function so you end up with the smaller functions having to return multiple variables which looks messy.